### PR TITLE
fix: unnecessary space on the left side

### LINKED ISSUE #33: chart reserves too much unnecessary space on the left side
## Scope
- To display numerical values of arbitrary length, svg-line chart reserves some fixed space on the left side.
- But when those numbers aren't big (in terms of the number of digits that they use), then when trying to center the resulting chart can look false as there's a lot of whitespace on the left side: see screenshot

Screen Shot 2021-11-04 at 15 43 22

## Deliverables

- Somehow measure the pixel length of the biggest number on the y axis and use minimal space on the left such that when the chart is centered in e.g. a container element it actually looks centered too.
- Another option would be to simply add the same amount of white space to the right side. That'd probably yield an easier solution but would be more difficult to integrate for a user later.

## Time estimate

- I'd guess that this issue is rather complex. I'd say an easy solution can take up to 4 hrs.

### PR DESCRIPTION
**Pull request by il3ven**
Fixes #33 

#### Solution:
The idea is to change the `offsetX` variable to the width of the `yLabel` instead of using a fixed value. This width is not in absolute pixel but relative to the given `viewBox`.

The function `getWidth` is based on the assumption that the `font-size` is kind of the height of the character and the width is proportional to the height. I couldn't find any spec for this but based on my tests I think it true at least for our use case.

I have tried inputting different values and the `getWidth` function holds. (See Tim's [comment](https://github.com/rugpullindex/svg-line-chart/pull/34#issuecomment-963244371))

#### Note
Due to this line we cannot specify the `fontSize` in em, cm or px. `1.5px` will throw an error. 
https://github.com/il3ven/svg-line-chart/blob/813b3d02812a968975d0eab28a6b006af0c2e146/src/index.mjs#L405

Do we need to normalize `1.5px` to `1.5` instead of throwing an error?

### CODE REVIEW

File: src/index.mjs [hunk 1]
```diff
@@ -398,3 +400,11 @@ export function generateLabelRange(min, max, numLabels) {
 
   return labels;
 }
+
+function getWidth(fontSize, dataPoints) {
+  const characterHeight = Number(fontSize) // This height is relative to viewbox and not in px
+  if(isNan(characterHeight)) throw new Error('Invalid fontSize')
```
Comment (TimDaub) [inline src/index.mjs:406]: I don't think `isNan` exists: https://developer.mozilla.org/de/docs/Web/JavaScript/Reference/Global_Objects/isNaN

**Comment by il3ven** [comment]
Yeah, I meant isNaN.

**Comment by TimDaub** [comment]
I took a few screenshots to document the behavior under different conditions
Screen Shot 2021-11-08 at 15 56 18
Screen Shot 2021-11-08 at 15 56 38
Screen Shot 2021-11-08 at 15 57 30
Screen Shot 2021-11-08 at 15 57 43
Screen Shot 2021-11-08 at 15 59 50
Screen Shot 2021-11-08 at 15 58 56

I think it's a good approach. I'm happy if we continue with it.

**Review by TimDaub — COMMENTED** [review]
Declare magic constants as variables.

I tend to prefer e.g calling the padding you added in the return by what it is `const padding = 2` and then later in the return use it.

File: src/index.mjs [hunk 2]
```diff
@@ -398,3 +400,11 @@ export function generateLabelRange(min, max, numLabels) {
 
   return labels;
 }
+
+export function getWidth(fontSize, dataPoints) {
+  const characterHeight = Number(fontSize) // This height is relative to viewbox and not in px
+  if(isNaN(characterHeight)) throw new Error('Invalid fontSize')
+  const characterWidth = characterHeight / 2 // For fonts such as Helvetica and Arial the ratio of height to width is ~2
+  const maxWidth = characterWidth * (String(Math.max(...dataPoints)).length)
+  return (maxWidth + 2) // Add padding
```
Comment (TimDaub) [inline src/index.mjs:409]: Declare magic constant. Also: Parenthesis not needed?

File: src/index.mjs [hunk 3]
```diff
@@ -398,3 +400,11 @@ export function generateLabelRange(min, max, numLabels) {
 
   return labels;
 }
+
+export function getWidth(fontSize, dataPoints) {
+  const characterHeight = Number(fontSize) // This height is relative to viewbox and not in px
+  if(isNaN(characterHeight)) throw new Error('Invalid fontSize')
+  const characterWidth = characterHeight / 2 // For fonts such as Helvetica and Arial the ratio of height to width is ~2
```
Comment (TimDaub) [inline src/index.mjs:407]: declare magic constant

Comment (il3ven) [reply src/index.mjs:407]: Makes sense. I'll do that.

File: src/index.mjs (hunk 2 above)
Comment (il3ven) [reply src/index.mjs:409]: This is a general problem. Can we put `.prettierrc` in the repo?

Comment (TimDaub) [reply src/index.mjs]: makes sense. u can create a new issue that can get paid separately and we leave the PR as is for now.

**Pull request merged**